### PR TITLE
Transform classic loops to range-based for loops in module tracking

### PR DESCRIPTION
**Pull request by SunBlack**
Changes are based on the result of run-clang-tidy -header-filter='.*' -checks='-*,modernize-loop-convert' -fix

### CODE REVIEW

**Comment by SunBlack** [comment]
I know, but I don't even have an idea why this methods are virtual, because they are just simple helper methods.

**Comment by SunBlack** [comment]
Same here: Just edited typo in commit message via force push (just saw that I copy & pasted a typo across my last PRs)

File: tracking/include/pcl/tracking/impl/kld_adaptive_particle_filter.hpp [hunk 1]
```diff
@@ -34,14 +34,14 @@ pcl::tracking::KLDAdaptiveParticleFilterTracker<PointInT, StateT>::initCompute (
...
 {
-  for (size_t i = 0; i < B.size (); i++)
+  for (auto &existing_bin : bins)
   {
-    if (equalBin (bin, B[i]))
+    if (equalBin (new_bin, existing_bin))
       return false;
   }
-  B.push_back (bin);
+  bins.push_back (std::move(new_bin));
```
Comment (SergioRAgostinho) [inline tracking/include/pcl/tracking/impl/kld_adaptive_particle_filter.hpp]: This `std::move` should be redundant. You're already declaring the argument as an rvalue reference in the function signature.

Comment (taketwo) [reply tracking/include/pcl/tracking/impl/kld_adaptive_particle_filter.hpp]: No it's not. Even if a variable’s type is an rvalue reference, the expression consisting of its name is an lvalue expression. So without `move` the `push_back` will receive an lvalue.

Comment (SergioRAgostinho) [reply tracking/include/pcl/tracking/impl/kld_adaptive_particle_filter.hpp]: :+1: I clearly spend too much time on python these days.

**Pull request merged**